conans/client/source.py

```diff
@@ -10,7 +10,8 @@
 from conans.model.conan_file import get_env_context_manager
 from conans.model.scm import SCM, get_scm_data
 from conans.paths import CONANFILE, CONAN_MANIFEST, EXPORT_SOURCES_TGZ_NAME, EXPORT_TGZ_NAME
-from conans.util.files import (clean_dirty, is_dirty, load, mkdir, rmdir, set_dirty, walk)
+from conans.util.files import (set_dirty, is_dirty, load, mkdir, rmdir, set_dirty_context_manager,
```

Comment:
do you still need to import `set_dirty`?

Reply:
yes, at line 125. In some places dirty_context is not the best choice.

Reply:
ok